src/providers/mod.rs

```diff
@@ -34,11 +34,19 @@ pub mod vagrant_virtualbox;
 
 use std::collections::HashMap;
 use std::fs::{self, File};
+#[cfg(not(feature = "legacy"))]
```

Comment:
If you want to avoid all these conditionals below, you can move most of the `use` statements directly inside function body.